crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -90,6 +92,24 @@ impl AccountBuilder {
         self
     }
 
+    /// Adds a designated authentication [`AccountComponent`] to the builder.
+    ///
+    /// This component will be placed at index 0 of the account procedures list.
+    pub fn with_auth_component(mut self, account_component: impl Into<AccountComponent>) -> Self {
```

Comment:
I think we should also add that (for the time being) this component should have exactly one procedure.

Reply:
Good point, fixed in [4c6219d](https://github.com/0xMiden/miden-base/pull/1480/commits/4c6219d7128bacbf41bc07709b89507e21dde128)